### PR TITLE
Add nf-test to pipeline template

### PR DESCRIPTION
**Pull request by edmundmiller**
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

We should probably update the docs somewhere :thinking:

### CODE REVIEW

File: nf_core/pipeline-template/.github/workflows/ci.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@ jobs:
         NXF_VER:
           - "21.10.3"
           - "latest-everything"
+        profile: ["docker"]
```
Comment (jfy133) [inline nf_core/pipeline-template/.github/workflows/ci.yml:29]: Shouldn't this also include `test` as we require this as the bare minimum? And shouldn't `docker` be in the command itself by default and the matrix only contains the actual test profiles?

**Review by jfy133 — COMMENTED** [review]
Some minor thoughts (not fully reviewed)

File: nf_core/pipeline-template/.github/workflows/ci.yml (hunk 1 above)
Comment (Emiller88) [reply nf_core/pipeline-template/.github/workflows/ci.yml:29]: The `test` profile is baked into nf-test now! In the matrix, it'll also include singularity and conda

**Comment by Emiller88** [comment]
Punting until the dust settles after the October Hackathon.

Might be good for Boston?

**Comment by mashehu** [comment]
Superseded by #2490
